### DISCUSSION TITLE
Canvas Containers?

### DISCUSSION THREAD

**Discussion by vnomus** [elsa-workflows/elsa-studio#174] [Q&A]
I created the elsa-studio (sever) and got it up and running ok in VS2022 and then built a msi installer for it. The application runs fine from VS and dotnet console. However, when run from the msi-installed instance, the activity containers don't render on the canvas. They're there as you can click on them, view properties, and the connecting lines show as well. I don't see any errors logged to the consoles(cmd and web-dev-tools). What am I missing?...

![image](https://github.com/elsa-workflows/elsa-studio/assets/1231006/7812ac3a-2c62-4e22-b316-5eb070ad344d)

**Comment by glime-ai** [comment]
To troubleshoot the issue you're experiencing with the MSI-installed instance of Elsa Studio, where activity containers don't render on the canvas, you can try the following steps:

1. **Check for JavaScript Errors**: Open the browser's developer tools (usually F12) and check the Console tab for any JavaScript errors that might be occurring when the page is loaded or when you try to interact with the workflow canvas.

2. **Inspect Network Activity**: In the developer tools, go to the Network tab and reload the page to ensure all resources are loading correctly. Look for any failed requests that might indicate missing files or resources.

3. **Review Application Logs**: If you have logging configured for your application, review the logs for any exceptions or errors that might provide clues about what's going wrong.

4. **Compare Configurations**: Compare the configuration files (e.g., `appsettings.json`) and environment settings between the development environment and the MSI-installed environment. There might be differences in configuration that affect the rendering of the activity containers.

5. **Check for Missing Dependencies**: Ensure that all necessary dependencies and static files are included in the MSI package. Sometimes, static files like CSS or JavaScript might not be included in the build, leading to rendering issues.

6. **Rebuild MSI with Debug Information**: If possible, rebuild the MSI with debug information and run the installed application in debug mode to step through the code and identify where the rendering process is failing.

7. **Consult Elsa Documentation and Community**: Check the Elsa documentation for any known issues related to deployment and rendering. You can also reach out to the Elsa community on forums or GitHub for assistance.

Since you mentioned that the application runs fine from Visual Studio and the dotnet console, the issue is likely related to the way the MSI package is built or configured. Ensuring that all necessary files are included and that the environment is set up correctly in the MSI-installed instance should help resolve the issue.

If you continue to face difficulties, please provide any error messages or additional details that could help in diagnosing the problem.
